docs/source/platform/operation-registry.mdx

```diff
@@ -273,6 +275,81 @@ const server = new ApolloServer({
 
 > _Note:_ The `forbidUnregisteredOperations` callback must be synchronous. If it is necessary to make an `async` request (e.g. a database inquiry) to make a determination about access, such a lookup should occur within the [`context` function](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#constructor-options-lt-ApolloServer-gt) on the `ApolloServer` constructor (or any life-cycle event which has access to `context`) and the result will be available on the `context` of `forbidUnregisteredOperations`.
 
+### Lifecycle methods
+
+The operation registry plugin supports three lifecycle methods to improve observability and modify the behavior of safelist: `willUpdateManifest`, `onUnregisteredOperation`, and `onForbiddenOperation`.
+
+#### willUpdateManifest(newManifest?: OperationManifest, oldManifest?: OperationManifest) => OperationManifest
+
+This function is run with `newManfifest` set to the result from gcs, when new manifest is returned by the poll.
+If the manifest is not retrieved from gcs on startup, `willUpdateManfiest` is run with `newManifest` set to `undefined`.
+Every time the function is called `oldManifest` set to the existing manifest if one is present.
+
+This function can be used to monitor the number of operations in the manifest and return a different or fallback manifest.
+
+<ExpansionPanel title="Code sample">
+
+```js{4-10}
+const server = new ApolloServer({
+  plugins: [
+    require("apollo-server-plugin-operation-registry")({
+      willUpdateManifest(newManifest, oldManifest) {
+        metrics.report(
+          "safelist.numberOperations",
+          newManifest.operations.length
+        );
+        return newManifest;
+      }
+    })
+  ]
+});
+```
+
+</ExpansionPanel>
+
+#### onUnregisteredOperation(requestContext)
+
+This function is run when an operation is not found in the safelist.
```

Comment:
weird that we're calling it the safelist instead of the manifest or registry. I see it as a fact of present implementation that those two are the same.